### PR TITLE
test: lift 16 below-85% files to ≥85% coverage (multi-agent orchestration)

### PR DESCRIPTION
**Pull request by jainal09**
## What

Raises unit-test coverage so **every source file is now ≥85%**. Overall project coverage **87.0% → 93.6%**.

Adds ~293 hermetic unit tests across **16 dedicated files** under `tests/unit/coverage/`, one per previously-sub-85% module. **No source files were modified** — tests only.

## Per-file results

| File | Before | After |
|------|-------:|------:|
| `cli_commands/install.py` | 69.7% | 97.4% |
| `scanner/kingfisher.py` | 76.5% | 98.0% |
| `cli_commands/hook.py` | 76.9% | 92.3% |
| `cli_commands/sync.py` | 77.5% | 87.2% |
| `scanner/engine.py` | 79.8% | 96.8% |
| `vault/hashicorp.py` | 80.6% | 100% |
| `scanner/infisical.py` | 81.5% | 98.1% |
| `scanner/trivy.py` | 82.0% | 98.5% |
| `cli_commands/encryption.py` | 82.3% | 96.7% |
| `scanner/talisman.py` | 82.7% | 96.3% |
| `cli_commands/agent.py` | 82.9% | 97.6% |
| `core/validator.py` | 83.5% | 99.5% |
| `encryption/base.py` | 83.9% | 100% |
| `integrations/hook_check.py` | 84.1% | 98.2% |
| `sync/engine.py` | 84.4% | 99.6% |
| `scanner/gitleaks.py` | 84.5% | 97.9% |

## How

Generated by a multi-agent **orchestration layer** (`.claude/workflows/coverage-orchestrator.js`): the main agent measured coverage and selected files below 85%, then fanned out — per file — a **test-writer** subagent, a parallel **SQA reviewer** (rejects filler, verifies the tests actually exercise the uncovered lines, hermetic, lint/type-clean), and an **indexer**. All 16 files were SQA-approved.

All tests are hermetic (external processes / vault clients / cloud SDKs mocked), carry no integration markers, and reuse existing fixtures/conventions.

## Verification

- ✅ `pytest -m "not integration"` — **2055 passed**, 0 failed
- ✅ docker integration suite (LocalStack + Vault + Lowkey) — **142 passed**, 0 failed
- ✅ pre-commit: ruff, ruff-format, pyrefly, bandit

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

# Release Notes

**No user-facing changes in this release.** This update includes internal infrastructure enhancements focused on test coverage expansion and development workflow improvements. These changes support system stability and code quality through comprehensive testing infrastructure without affecting user functionality.

<h3>Greptile Summary</h3>

Adds ~293 hermetic unit tests across 16 dedicated files under `tests/unit/coverage/` to lift every previously-below-85% source file to ≥85% coverage, raising overall project coverage from 87.0% to 93.6%. No source files are modified; all external processes, vault clients, and cloud SDKs are mocked.

- **Test structure**: Each file targets a specific source module's uncovered branches (error handlers, import-fallbacks, platform guards, edge cases). Parametrised tests cover all 8 optional scanner ImportError and successful-init paths. Real `tarfile`/`zipfile` archives are used where path-traversal guards are tested, giving stronger fidelity than pure mocking.
- **`conftest.py` gap**: The autouse `_force_wide_console` fixture correctly pins Rich console widths to prevent CI flakiness from output wrapping, but `envdrift.cli_commands.sync` — which has a module-level `console` and is the subject of the largest file in this suite — is missing from `_CONSOLE_MODULES`.

<h3>Confidence Score: 5/5</h3>

Test-only PR with no source changes; all tests are hermetic and reportedly pass at 2055/0. Safe to merge.

All changes are in the test layer with no modifications to production code. Mocking strategies are appropriate for the lazy-import patterns used in the source modules. The one gap (missing sync module from the console-width fixture list) is a low-risk flakiness vector, not a correctness issue.

tests/unit/coverage/conftest.py — `_CONSOLE_MODULES` should include `envdrift.cli_commands.sync` to match the fixture's documented intent.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| tests/unit/coverage/conftest.py | Autouse `_force_wide_console` fixture correctly addresses Rich console width flakiness, but `_CONSOLE_MODULES` is missing `envdrift.cli_commands.sync`, the module under test by the largest file in this suite. |
| tests/unit/coverage/test_cov_cli_commands_sync.py | Extensive coverage of sync CLI branches including profile filtering, partial encryption, rekey paths, and interactive prompts. Mocking is correct (SyncEngine is lazily imported per-function in the source, so patching at `envdrift.sync.engine.SyncEngine` works). |
| tests/unit/coverage/test_cov_vault_hashicorp.py | Module-level import now hoisted. `test_import_without_hvac_sets_unavailable` guards correctly with `pytest.importorskip("hvac")`, cleanly exercising the ImportError fallback block without polluting typed exception aliases for subsequent tests. |
| tests/unit/coverage/test_cov_scanner_engine.py | Well-structured parametrised tests for all 8 optional scanners' ImportError and successful-init branches. Edge cases for timeout, gitignore filtering, and line-at helpers are hermetic and clean. |
| tests/unit/coverage/test_cov_sync_engine.py | Good coverage of SyncEngine branches. The `_patch_folder_missing` approach of globally patching `Path.exists` is handled safely by monkeypatch teardown. |
| tests/unit/coverage/test_cov_scanner_gitleaks.py | Thorough coverage of download/extract errors, path-traversal guards, and scan error paths. Uses real `tarfile`/`zipfile` archives for stronger fidelity. |
| tests/unit/coverage/test_cov_encryption_base.py | Uses `_ =` captures instead of `assert ... is None` checks, making abstract-body coverage tests resilient to future changes. |
| tests/unit/coverage/test_cov_core_validator.py | Covers missing-optional detection, float/bool/list/empty type-check branches, and fix-template generation. Hermetic and free of external dependencies. |
| tests/unit/coverage/test_cov_integrations_hook_check.py | Covers absolute-path resolution, inject-block newline branches, chmod OSError fallback, and walk-to-root logic. Well-scoped. |
| .claude/workflows/coverage-orchestrator.js | Orchestration script documenting the multi-agent workflow used to generate the tests. Not part of production codebase. |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[coverage-orchestrator.js] --> B[Measure coverage per file]
    B --> C{File < 85%?}
    C -- Yes --> D[Fan out per file]
    D --> E[test-writer subagent]
    D --> F[SQA reviewer subagent]
    D --> G[indexer subagent]
    E --> H[Hermetic unit tests\nall external calls mocked]
    F --> I{SQA approved?}
    I -- Yes --> J[tests/unit/coverage/\ntest_cov_*.py]
    I -- No --> E
    G --> K[Coverage index updated]
    J --> L[pytest -m not integration\n2055 passed]
    C -- No --> M[Skip file]
    J --> N[conftest.py\n_force_wide_console autouse fixture]
    N --> O[Pin Rich consoles wide\nbefore each test]
```

<sub>Reviews (4): Last reviewed commit: ["test: address greptile P2 review feedbac..."](https://github.com/jainal09/envdrift/commit/c435c76d0928652984b5ad9015af8931bbddf211) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=35647228)</sub>

### CODE REVIEW

**Comment by jainal09** [comment]
All CI checks are green ✅ (Tests & Coverage, Integration Tests on Python 3.11–3.14, codecov, CodeQL, CodeRabbit, benchmarks).

Disposition of automated review feedback:

- **CodeQL — incomplete URL substring sanitization** (`test_cov_cli_commands_encryption.py`): fixed in `d4a5c89` — the SOPS kms kwargs are now asserted by exact equality instead of `str().startswith(...)`. CodeQL now passes.
- **greptile P2 — `MagicMock` imported per-method** (`test_cov_vault_hashicorp.py`): fixed in `d4a5c89` — hoisted to a module-level import, matching the other coverage test files.
- **CI flakiness** (3 `lock` tests failed only in the full CI suite): fixed in `aa382c7` — root-caused to the shared module-level Rich `console` being left at a narrow width by another test, wrapping output mid-phrase and breaking substring assertions. Added an autouse wide-console fixture in `tests/unit/coverage/conftest.py`. Reproduced and verified.
- **greptile P2 — module-reload hygiene & abstract-body assertions**: left intentionally. These exercise the `ImportError` fallback and the abstract-method bodies specifically for line coverage; the reload-in-`finally` is the intended cleanup and the abstract-body assertions are deliberate.

Net: overall coverage 87.0% → 93.6%, every source file now ≥85%.

File: tests/unit/coverage/test_cov_vault_hashicorp.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+from __future__ import annotations
+
+import builtins
+import importlib
+
+import pytest
+
+import envdrift.vault.hashicorp as hashicorp
+from envdrift.vault.base import AuthenticationError, VaultError
+from envdrift.vault.hashicorp import HashiCorpVaultClient
```
[the remark this answers is not in the document]
Comment (jainal09) [reply tests/unit/coverage/test_cov_vault_hashicorp.py:25]: Fixed in d4a5c89 — hoisted `from unittest.mock import MagicMock` to a module-level import. Resolving.

File: tests/unit/coverage/test_cov_vault_hashicorp.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+class TestAuthenticateErrorBranches:
+    """Cover authenticate() and is_authenticated() error handling."""
+
+    def test_authenticate_unauthorized_raises_auth_error(
+        self, monkeypatch: pytest.MonkeyPatch
+    ) -> None:
+        """Unauthorized during Client construction -> AuthenticationError (line 89)."""
+        from unittest.mock import MagicMock
+
+        mock_hvac = MagicMock()
```
[the remark this answers is not in the document]
Comment (jainal09) [reply tests/unit/coverage/test_cov_vault_hashicorp.py:118]: Fixed in c435c76 — guarded this test with `pytest.importorskip("hvac")` so the `finally` reload only runs when hvac is importable and restores the real typed sentinels. Resolving.

File: tests/unit/coverage/test_cov_encryption_base.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        return "ENC:" in content
+
+    def install_instructions(self) -> str:
+        # Executes the abstract ``install_instructions`` body (line 208).
+        assert EncryptionBackend.install_instructions(self) is None
+        return "install supercaller"
+
+
+def test_abstract_property_bodies_run() -> None:
+    """Calling the super property getters runs lines 63 and 75."""
```
[the remark this answers is not in the document]
Comment (jainal09) [reply tests/unit/coverage/test_cov_encryption_base.py]: Fixed in c435c76 — replaced the `assert ... is None` side-checks with `_ =` captures; the abstract bodies are still executed for coverage (envdrift.encryption.base stays 100%) without coupling to their return. Resolving.
